test/Elastic.Apm.Tests/BasicAgentTests.cs

```diff
@@ -30,5 +38,30 @@ public void AgentVersion()
 
 			Assert.Equal(Assembly.Load("Elastic.Apm").GetName().Version.ToString(), payloadSender.Payloads[0].Service.Agent.Version);
 		}
+
+		[Fact]
+		public void PayloadSentWithBearerToken()
+		{
+			AuthenticationHeaderValue authHeader = null;
+			var handler = new MockHttpMessageHandler((r, c) =>
+			{
+				authHeader = r.Headers.Authorization;
+				return Task.FromResult(new HttpResponseMessage(HttpStatusCode.OK));
+			});
+
+			var secretToken = "SecretToken";
+			var logger = ConsoleLogger.Instance;
+			var payloadSender = new PayloadSender(logger, new TestAgentConfigurationReader(logger, secretToken: secretToken), handler);
+
+			using (var agent = new ApmAgent(new TestAgentComponents(secretToken: secretToken, payloadSender: payloadSender)))
+				agent.PayloadSender.QueuePayload(new Payload());
+
+			// ideally, introduce a mechanism to flush payloads
```

Comment:
Agree :) It's ok for now, in #13 there will be more changes in `PayloadSender`, I'll also add this functionality.